js_modules/dagit/packages/core/src/assets/AssetNodeLineage.tsx

```diff
@@ -65,3 +101,57 @@ export const AssetNodeLineage: React.FC<{
     </>
   );
 };
+
+const LineageDepthControl: React.FC<{
+  value: number;
+  max: number;
+  onChange: (v: number) => void;
+}> = ({value, max, onChange}) => {
+  const commitValue = (target: HTMLInputElement) => {
+    const next = Number(target.value) ? Math.min(max, Number(target.value)) : value;
+    onChange(next);
+    target.value = `${next}`;
```

Comment:
Ahh man... I debated making it a controlled input, but we'd need a separate piece of state for the value because I want it to be ok to erase the value (to "") and then type in another value without it inserting a `0`. Committing on blur / enter and normalizing the value is definitely a bit of a hack though... let me try the controlled version again and see if I can make it nice.